slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -224,6 +225,109 @@ void SLAMServiceImpl::SetUpMapBuilder() {
 }
 
 std::string SLAMServiceImpl::GetLatestJpegMapString(bool add_pose_marker) {
+    
+    cartographer::io::PaintSubmapSlicesResult painted_slices = GetLatestPaintedMapSlices();
+    if (add_pose_marker) {
+        PaintMarker(painted_slices);
+    }
+
+    auto image = io::Image(std::move(painted_slices.surface));
+    return image.WriteJpegToString(50);
+}
+
+bool SLAMServiceImpl::GetLatestPointCloudMapString(std::string &pointcloud) {
+    
+    cartographer::io::PaintSubmapSlicesResult painted_slices = GetLatestPaintedMapSlices();
+    auto painted_surface = painted_slices.surface.get();
+    int width = cairo_image_surface_get_width(painted_surface);
+    int height = cairo_image_surface_get_height(painted_surface);
+
+    //total number of bytes, 32 bits per pixel
+    size_t size_data = width*height*4;
+    auto data = cairo_image_surface_get_data(painted_surface);
+
+    std::vector<unsigned char> data_vect(data, data + size_data);
+    
+    std::stringbuf data_buffer;
+
+    int num_points = 0;
+
+    // Reduce resolution based off number of pixels. Output is pixels to skip
+    int scaleFactor = 4;
+    int scale = size_data/100000.f/32.f/scaleFactor;
+    
+    // Loop to filter unwanted data and reduce resolution
+    for (int i=0;i<size_data;i=i+scale*4) {
+        int rgb = 0;
+        rgb = rgb | ((int)data_vect[i+0] << 16);
+        rgb = rgb | ((int)data_vect[i+1] << 8);
+        rgb = rgb | ((int)data_vect[i+2] << 0);
+        
+        //skip pixels that are not in our map(black/past walls)
+        // this value represents [102,102,102]
+        if(rgb == 6710886)
+        continue;
+
+        //Determine probability based off color pixel
+        int prob = ViamColorToProbability((int)data_vect[i+2]);
+        
+        if(prob == 0)
+        continue;
+        
+        num_points++;
+        
+        int pixel_index = i/4;
+        int pixel_x = pixel_index%width;
+        int pixel_y = pixel_index/width;
+
+        // kPixelSize is .01, unsure of units but assumed converts to meters
+        // based on PaintSubmapSlices() and DrawPoseOnSurface()
+        float x_pos = (pixel_x-painted_slices.origin.x())*kPixelSize;
+        // Y is inverted to match output from getPosition()
+        float y_pos = -(pixel_y-painted_slices.origin.y())*kPixelSize;
+        // 2D SLAM so Z is set to 0
+        float z_pos = 0;
+
+        //rotating coordinates to match slam service expectation(XZ plane)
+        data_buffer.sputn((const char *)&x_pos, 4);
+        data_buffer.sputn((const char *)&z_pos, 4);
+        data_buffer.sputn((const char *)&y_pos, 4);
```

Comment:
Realized during this PR that switching the Y and Z fields here will cause a misalignment with the `GetPosition` endpoint

Reply:
update: still looking into updating GetPosition in this PR. Will comment back here when its added

Reply:
added in the rotation, but integration tests fail due to the getPosition test no longer matching. Will make a quick PR in rdk to get integration tests to pass